File: MyApplication5/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.InputStream;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private TextView titleTextView, genreTextView, yearTextView, directorTextView, actorsTextView, ratingTextView;
    private Button randomMovieButton;
    private JSONArray moviesArray;
    private List<Integer> movieIndices;
    private int lastRandomMovieIndex;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        titleTextView = findViewById(R.id.titleTextView);
        genreTextView = findViewById(R.id.genreTextView);
        yearTextView = findViewById(R.id.yearTextView);
        directorTextView = findViewById(R.id.directorTextView);
        actorsTextView = findViewById(R.id.actorsTextView);
        ratingTextView = findViewById(R.id.ratingTextView);
        randomMovieButton = findViewById(R.id.randomMovieButton);

        try {

            InputStream is = getResources().openRawResource(R.raw.movies);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();

            String json = new String(buffer, "UTF-8");
            JSONObject jsonObject = new JSONObject(json);
            moviesArray = jsonObject.getJSONArray("movies");
        } catch (Exception e) {
            e.printStackTrace();
        }


        movieIndices = new ArrayList<>();
        for (int i = 0; i < moviesArray.length(); i++) {
            movieIndices.add(i);
        }
        Collections.shuffle(movieIndices);

        randomMovieButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                int randomIndex = generateRandomIndex();

                try {

                    JSONObject randomMovie = moviesArray.getJSONObject(randomIndex);
                    lastRandomMovieIndex++;

                    String title = randomMovie.getString("title");
                    String genre = randomMovie.getString("genre");
                    int year = randomMovie.getInt("year");
                    String director = randomMovie.getString("director");
                    JSONArray actorsArray = randomMovie.getJSONArray("actors");
                    double rating = randomMovie.getDouble("rating");


                    titleTextView.setText("Название: " + title);
                    genreTextView.setText("Жанр: " + genre);
                    yearTextView.setText("Год: " + year);
                    directorTextView.setText("Режиссер: " + director);


                    StringBuilder actorsStringBuilder = new StringBuilder("Актеры: ");
                    for (int i = 0; i < actorsArray.length(); i++) {
                        actorsStringBuilder.append(actorsArray.getString(i));
                        if (i < actorsArray.length() - 1) {
                            actorsStringBuilder.append(", ");
                        }
                    }
                    actorsTextView.setText(actorsStringBuilder.toString());

                    ratingTextView.setText("Рейтинг: " + rating);
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });
    }

    private int generateRandomIndex() {
        int arrayLength = movieIndices.size();


        if (lastRandomMovieIndex >= arrayLength) {
            Collections.shuffle(movieIndices);
            lastRandomMovieIndex = 0;
        }

        int randomIndex = movieIndices.get(lastRandomMovieIndex);
        return randomIndex;
    }
}

File: MyApplication6/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editTextRange: EditText = findViewById(R.id.editTextRange)
        val buttonStartGame: Button = findViewById(R.id.buttonStartGame)

        buttonStartGame.setOnClickListener {
            val range = editTextRange.text.toString()

            if (isValidRange(range)) {
                val (rangeStart, rangeEnd) = parseRange(range)
                if (rangeStart < rangeEnd) {
                    val intent = Intent(this, GuessingActivity::class.java)
                    intent.putExtra("rangeStart", rangeStart)
                    intent.putExtra("rangeEnd", rangeEnd)
                    startActivity(intent)
                } else {
                    Toast.makeText(
                        this,
                        "Некорректный диапазон. Первое число должно быть меньше второго.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            } else {
                Toast.makeText(
                    this,
                    "Некорректный формат диапазона. Используйте формат, например, 1-100.",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }

    private fun isValidRange(range: String): Boolean {
        val regex = """^\d+-\d+$""".toRegex()
        return range.matches(regex)
    }

    private fun parseRange(range: String): Pair<Int, Int> {
        val (start, end) = range.split("-")
        return Pair(start.toInt(), end.toInt())
    }
}
